rfcs/text/0000-rfc-application-layer-bundle-types-and-definitions.md

```diff
@@ -0,0 +1,434 @@
+
+
+1. As a Data Consumer, I would like to work in the natural model of the domain without having to learn data storage system related concepts.
+
+
+1. As a Data Consumer, I would like to be able to tell the type of a bundle by inspecting the domain metadata contents rather than examining storage system metadata. 
+
+
+1. As a Data Consumer, I would like key type parameters to be more explicit in the metadata so they can be discovered by inspection. 
+
+
+1. As a Developer or Data Consumer, I would like clear documentation enumerating bundle types explaining their shape and purpose.
+
+
+## Scientific "guardrails" [optional]
+
+*Describe recommended or mandated review from HCA Science governance to ensure that the RFC addresses the needs of the scientific community.*
+
+## Detailed Design
+
+### Overview
+
+To address the above motivation, it is proposed to:
+
+1. Create an extensible type system with five main types: _Assay_, _Analysis_, _Reference_, _Entity_ and _Event_.
+
+
+1. Allow for one level of subtyping to refine the types.
+
+
+1. Further refine the type by referencing the process and protocol, used to create the bundle and other type specific information as required.
+
+
+1. Add a type.json file to each bundle to specify the bundle type information.
+
+
+1. Specify the desired types in JSON schema.
+
+
+1. Document bundle types and shapes in the data portal.
+
+#### Why JSON and JSON Schema instead of media type syntax?
+JSON is more human readable, and can express more complex types and can be validated. Also we are significantly invested in JSON schema and have the tools and processes to support the schema lifecycle in place.
+
+
+### Core Types
+
+![Core Types](../images/0000-rfc-application-layer-bundle-types-coretypes.png)
+
+#### Assay Bundle
+
+Umbrella type for Assays/Measurements with the following subclasses:
+
+
+1. “Sequencing Assay” - representing the current sequencing ingest bundles.
+
+1. “Imaging Assay” - representing the current imaging ingest bundles. 
+
+1. Other high level assay types.
+
+
+> _Note: It may take some discussion to decide on an effective set of subclasses. Sequencing assay may need to change to “2nd Gen Sequencing Assay” or “NGS Sequencing Assay” for example._
+
+#### Analysis Bundle
+Umbrella type used to represent an analysis with the following possible subclasses:
+
+
+1. “Alignment-Expression” - representing our current secondary analysis pipelines.
+
+
+1. “Expression Matrix” - a possible subtype representing the expression matrix generation.
+
+
+1. “Expression Normalization” - a possible subtype representing the expression matrix normalization output for example.
+
+
+1. “Clustering” - a possible subtype representing cell clustering analyses output for example.
+
+> _Note: The exact names of the subclasses above can likely be improved upon. Please suggest._
+
+#### Reference Bundle
+Umbrella type representing a reference file used in analysis. Subclasses TBD.
```

Comment:
I think we will have a better feel for this as we get more data modalities in the system. For now, this may be as simple as human and mouse. It may be good not to use the common name but use some specific ontology for the species names (I am sure our EBI friends have a favorite ontology to use). @lauraclarke